libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -351,16 +350,15 @@ def _nodes_with_ids(
         self,
         ids: Iterable[str],
     ) -> List[Node]:
-        results = {}
+        results: Dict[str, Node] = {}
         with self._concurrent_queries() as cq:
 
-            def add_nodes(rows):
+            def add_nodes(rows: Iterable[Any]) -> None:
                 for row in rows:
                     results[row.content_id] = _row_to_node(row)
 
             for node_id in ids:
                 if node_id not in results:
-                    results[node_id] = None
```

Comment:
@bjchambers this complicates the typing and doesn't seem needed. So I removed it. OK for this ?

Reply:
We may want this, since it changes the behavior of the function on duplicate IDs. We need to either verify we won't call it with duplicates, or document that it will issue multiple requests in that case. Or, we should leave as is. I don't think it should complicate the typing excessively. I would suggest:

1. `results: Dict[str, Optional[Node]]`
2. `return [results[node_id] for node_id in ids if results[node_id] is not None]`. This could also be an assertion, since it being `None` means we requested an invalid ID. Alternatively, we may want to handle invalid ID in the `add_nodes` case (an empty `rows` indicates a problem).

The result type should still be `Node`.

Reply:
I've put it back. The handling of None values (or empty resultset in the callback) can be done in a subsequent PR.